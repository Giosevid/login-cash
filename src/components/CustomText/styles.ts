import {StyleSheet, ViewStyle} from 'react-native';

type StyleProps = {
  regular: ViewStyle;
  bold: ViewStyle;
  light: ViewStyle;
};

const styles = StyleSheet.create<StyleProps>({
  regular: {
    fontWeight: 'normal',
  },
  bold: {
    fontWeight: 'bold',
  },
  light: {
    fontWeight: '100',
  },
});

export default styles;
