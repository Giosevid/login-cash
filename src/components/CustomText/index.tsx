import React, {FC} from 'react';
import {Text, TextStyle} from 'react-native';
import styles from './styles';

type CustomTextProps = {
  textType?: 'regular' | 'bold' | 'light';
  textStyle?: TextStyle | TextStyle[];
};

const CustomText: FC<CustomTextProps> = ({
  children,
  textStyle,
  textType = 'regular',
}) => {
  const applyStyle = {
    regular: styles.regular,
    bold: styles.bold,
    light: styles.light,
  };

  return <Text style={[textStyle, applyStyle[textType]]}>{children}</Text>;
};

export default CustomText;
