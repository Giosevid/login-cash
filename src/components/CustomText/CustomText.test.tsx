import React from 'react';
import {render} from '@testing-library/react-native';
import CustomText from './index';

describe('Test Button', () => {
  const cases = ['regular', 'bold', 'light'];

  test.each(cases)('expect textType with %p and returns %p', firstArg => {
    const {toJSON} = render(
      <CustomText children="Test Button" textType={firstArg} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
