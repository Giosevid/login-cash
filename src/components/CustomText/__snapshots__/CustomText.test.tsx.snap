// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button expect textType with "bold" and returns %p 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontWeight": "bold",
      },
    ]
  }
>
  Test Button
</Text>
`;

exports[`Test Button expect textType with "light" and returns %p 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontWeight": "100",
      },
    ]
  }
>
  Test Button
</Text>
`;

exports[`Test Button expect textType with "regular" and returns %p 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontWeight": "normal",
      },
    ]
  }
>
  Test Button
</Text>
`;
